crates/wysiwyg/src/dom/nodes/container_node.rs

```diff
@@ -198,6 +231,64 @@ where
             _ => false,
         }
     }
+
+    pub fn kind(&self) -> ContainerNodeKind<S> {
+        self.kind.clone()
+    }
+
+    pub fn is_same_kind(&self, other: &ContainerNode<S>) -> bool {
```

Comment:
If I understand this correctly and its usage within `merge_children`, shouldn't we rather have some kind of `is_mergeable_with` func, as I don't expect we want to merge nodes like Links, ListItems, etc

Reply:
You might be right, right now on `NodeJoiner` we check that tags are the same and I thought we could replace that with this `is_same_kind` function, but a more specialised version could be better for both cases.